### ISSUE TITLE
On 32-bit, usize as f64 recommends f64::from(usize)

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#3689]
On i686, the cast_lossless lint recommends using `f64::from(len)` instead of `len as f64` where `len` is of type `usize`. This would <del>change the behaviour on x86_64</del> fail to compile.

**Comment by repi** [comment]
We ran into this as well when compiling to `wasm32-unknown-unknown`:

```sh
warning: casting `usize` to `f64` may become silently lossy if you later change the type
  --> world-benchmark/src/lib.rs:94:28
   |
94 |                         / (self.ns_history.len() as f64 * f64::from(n))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `f64::from(self.ns_history.len())`
   |
   = note: requested on the command line with `-W clippy::cast-lossless`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_lossless
```

following this suggestion leads to:

```
error[E0277]: the trait bound `f64: std::convert::From<usize>` is not satisfied
  --> world-benchmark/src/lib.rs:94:28
   |
94 |                         / (f64::from(self.ns_history.len()) * f64::from(n))
   |                            ^^^^^^^^^ the trait `std::convert::From<usize>` is not implemented for `f64`
   |
   = help: the following implementations were found:
             <f64 as std::convert::From<f32>>
             <f64 as std::convert::From<i16>>
             <f64 as std::convert::From<i32>>
             <f64 as std::convert::From<i8>>
           and 135 others

For more information about this error, try `rustc --explain E0277`.
```

would be great fox fix this in the lint, we do find it useful and works well for all other code we have.
